## Five Key References

1. Stephanie C Hicks, F William Townes, Mingxiang Teng, Rafael A Irizarry. 
Missing data and technical variability in single-cell RNA-sequencing experiments.
Biostatistics (2017-11-06) https://doi.org/gfb8g4
DOI: 10.1093/biostatistics/kxx053 · PMID: 29121214 · PMCID: PMC6215955

2. Genevieve L Stein-O’Brien, Brian S. Clark, Thomas Sherman, Christina Zibetti, Qiwen Hu, Rachel Sealfon, Sheng Liu, Jiang Qian, Carlo Colantuoni, Seth Blackshaw, Loyal A. Goff, Elana J. Fertig. 
Decomposing cell identity for transfer learning across cellular measurements, platforms, tissues, and species.
Cold Spring Harbor Laboratory (2018-08-20) https://doi.org/gd2xpn
DOI: 10.1101/395004

3. Avi Srivastava, Laraib Malik, Tom Sean Smith, Ian Sudbery, Rob Patro. 
Alevin efficiently estimates accurate gene abundances from dscRNA-seq data. 
Cold Spring Harbor Laboratory (2018-06-01) https://doi.org/gffk42
DOI: 10.1101/335000

4. Brian Clark, Genevieve Stein-O’Brien, Fion Shiau, Gabrielle Cannon, Emily Davis, Thomas Sherman, Fatemeh Rajaii, Rebecca James-Esposito, Richard Gronostajski, Elana J. Fertig, Loyal A. Goff, Seth Blackshaw. 
Comprehensive analysis of retinal development at single cell resolution identifies NFI factors as essential for mitotic exit and specification of late-born cells. 
Cold Spring Harbor Laboratory (2018-07-27) https://doi.org/gdwrzh
DOI: 10.1101/378950

5. Rob Patro & Michael Love. 
tximeta. 
Bioconductor (2018) https://doi.org/gfddxw
DOI: 10.18129/b9.bioc.tximeta
