## Project Team

### PI information

1. Loyal Goff (Submitter)

    * Title: Assistant Professor
    * Degrees: PhD
    * Type of organization: Academic
    * Tax ID: 52-0595110 (JHU)
    * Email: loyalgoff@jhmi.edu

2. Stephanie Hicks

    * Title: Assistant Professor
    * Degrees: PhD
    * Type of organization: Academic
    * Tax ID: 52-0595110 (JHU)
    * Email: shicks19@jhu.edu

3. Elana Fertig

    * Title: Associate Professor
    * Degrees: PhD
    * Type of organization: Academic
    * Tax ID: 52-0595110 (JHU)
    * Email: ejfertig@jhmi.edu

4. Casey Greene

    * Title: Assistant Professor
    * Degrees: PhD
    * Type of organization: Academic
    * Tax ID: 23-1352685 (UPenn)
    * Email: greenescientist@gmail.com

5. Thomas Hampton

    * Title: Senior Bioinformatics Analyst
    * Degrees: PhD
    * Type of organization: Academic
    * Tax ID: 02-0222111 (Dartmouth)
    * Email: Thomas.H.Hampton@dartmouth.edu

6. Michael Love

    * Title: Assistant Professor
    * Degrees: Dr. rer. nat.
    * Type of organization: Academic
    * Tax ID: 56-6001393 (UNC)
    * Email: milove@email.unc.edu

7. Rob Patro
    * Title: Assistant Professor
    * Degrees: PhD
    * Type of Organization: Academic
    * Tax ID: 16-1514621 (Stony Brook)
    * Email: rob.patro@cs.stonybrook.edu


### Description (750 words TOTAL)

1. Loyal Goff
2. Stephanie C. Hicks is an Assistant Professor of Biostatistics at the Johns Hopkins Bloomberg School of Public Health. She is an expert in statistical methodology with a strong track record in processing and analyzing single-cell genomics data, including extensive experience developing fast, memory-efficient R/Bioconductor software to remove systematic and technical biases from scRNA-seq data [@doi:10.1093/biostatistics/kxx053]. Dr. Hicks will work together with Co-PIs to implement fast search algorithms in latent spaces (Aim 1) and to implement the methods developed into fast, scalable, and memory-efficient R/Bioconductor software packages (Aim 3).
3. Elana Fertig is an Associate Professor of Oncology and Applied Mathematics and Statistics at Johns Hopkins University. She developed of the Bayesian non-negative matrix factorization algorithm CoGAPS [@doi:10.1093/bioinformatics/btq503] for latent space analysis. In collaboration with co-PI Goff, she adapted this tool to scRNA-seq data and developed a new transfer learning framework to relate the low-dimensional features in scRNA-seq data across data modalities, biological conditions, and organisms [@doi:10.1101/395004]. Dr. Fertig will work with the co-PIs to incorporate the error models from Aim 1 into the latent space representations, dimensionality estimation, and biological assessment metrics in Aim 2. She is developing standardized language for latent space representation in collaboration with co-PIs Goff and Greene [@doi:10.1016/j.tig.2018.07.003] that will provide a strong foundation for standardization of these approaches across different unsupervised learning tools.
4. Casey Greene is an Assistant Professor of Systems Pharmacology and Translational Therapeutics at the University of Pennsylvania's Perelman School of Medicine. He is an expert in deep learning techniques that learn low-dimensional representations of gene expression data[@doi:10.1128/mSystems.00025-15,@doi:10.1016/j.cels.2017.06.003,@pmid:29218871,@doi:10.1101/385534]. He will work with the co-PIs to implement and evaluate techniques that learn shared low-dimensional representations for scRNA-seq data and methods to search over them (Aim 1). He has experience teaching machine learning to non-computational biologists, including at a course that he has worked with co-PI Tom Hampton on, and he will enhance and extend this curriculum to support machine learning methods over the HCA in this proposal (Aim 3).
5. Tom Hampton is Director of Bioinformatic Training for two program projects at the Geisel School of Medicine at Dartmouth. In that role, he has a long collaboration with co-PI Casey Greene, with whom he has collaborated in the development of short courses taught at Mount Desert Island Biological Laboratory and at Dartmouth. Dr Hampton’s  bioinformatic research is focused on using data from multiple independent studies to identify concordant patterns of gene express in response to stressors such as infection and environmental stress.
6. Michael Love is an Assistant Professor of Biostatistics and Genetics at the University of North Carolina at Chapel Hill. He is a leading developer of statistical software for RNA-seq analysis in the Bioconductor Project, maintaining the widely used DESeq2 [@doi:10.1186/s13059-014-0550-8] and tximport [@doi:10.12688/f1000research.7563.1] packages. He is a close collaborator with Dr. Rob Patro on bias-aware estimation of transcript abundance from RNA-seq and estimation of uncertainty during transcript quantification [@doi:10.1038/nmeth.4197]. Dr. Love will work with co-PIs to disseminate versioned reference cell type catalogs through widely used frameworks for genomic data analysis including R/Bioconductor and Python.
7. Rob Patro is an Assistant Professor of Computer Science at Stony Brook University.  He leads the COMBINE-lab, that [develops and maintains numerous open-source genomics tools and methods](https://github.com/COMBINE-lab).  He is the primary developer of the popular transcript quantification tools Sailfish [@doi:10.1038/nbt.2862] and Salmon [@doi:10.1038/nmeth.4197], having collaborated closely with Dr. Love on the latter.  Dr. Love and he are actively collaborating on improved methods for transcript quantification, differential testing, and also on reproducible analysis via [tximeta](https://github.com/mikelove/tximeta) [@doi:10.18129/B9.bioc.tximeta]. He has recently focused on developing improved methods for gene-level quantification from tagged-end single-cell RNA-seq data, as implemented in the tool alevin [@doi:10.1101/335000]. He will work with co-PIs to develop improved single-cell quantification tools that account for gene-ambiguous reads and provide quantification uncertainty estimates (base enabling technologies) --- which is important for accurate and robust creation of reduced-dimensionality representations.  He will work with the co-PIs to develop efficient algorithms and data structures to enable efficient expression and sample search over low-dimensional representations of HCA data (Aim 1).
