<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Loyal A. Goff" />
  <meta name="author" content="Casey S. Greene" />
  <meta name="author" content="Stephanie C. Hicks" />
  <meta name="author" content="Rob Patro" />
  <meta name="author" content="Elana J. Fertig" />
  <meta name="author" content="Michael I. Love" />
  <meta name="dcterms.date" content="2018-11-10" />
  <meta name="keywords" content="dimensionality reduction, search, transformation, reference, cell types, single cell, Human Cell Atlas" />
  <title>Practical search and analysis with low-dimensional representations of the HCA</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="github-pandoc.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!-- Insert Analytics Script Below -->
  <script>
  </script>
  <!-- End Analytics Script -->
</head>
<body>
<header>
<h1 class="title">Practical search and analysis with low-dimensional representations of the HCA</h1>
</header>
<p><small><em> This manuscript (<a href="https://greenelab.github.io/czi-seed-rfa/v/07afb950f7edb9d15d11668acf5c2376d9a4d88f/">permalink</a>) was automatically generated from <a href="https://github.com/greenelab/czi-seed-rfa/tree/07afb950f7edb9d15d11668acf5c2376d9a4d88f">greenelab/czi-seed-rfa@07afb95</a> on November 10, 2018. </em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Loyal A. Goff</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0003-2875-451X">0000-0003-2875-451X</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/loyale">loyale</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/loyalgoff">loyalgoff</a><br> <small> Solomon H. Snyder Department of Neuroscience, Johns Hopkins University School of Medicine; Kavli Neurodiscovery Institute, Johns Hopkins University; McKusick-Nathans Institute of Genetic Medicine, Johns Hopkins University School of Medicine </small></p></li>
<li><p><strong>Casey S. Greene</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0001-8713-9213">0000-0001-8713-9213</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/cgreene">cgreene</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/greenescientist">greenescientist</a><br> <small> Department of Systems Pharmacology and Translational Therapeutics, Perelman School of Medicine, University of Pennsylvania </small></p></li>
<li><p><strong>Stephanie C. Hicks</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0002-7858-0231">0000-0002-7858-0231</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/stephaniehicks">stephaniehicks</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/stephaniechicks">stephaniechicks</a><br> <small> Department of Biostatistics, Johns Hopkins Bloomberg School of Public Health </small></p></li>
<li><p><strong>Rob Patro</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0001-8463-1675">0000-0001-8463-1675</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/rob-p">rob-p</a><br> <small> Department of Computer Science, Stony Brook University </small></p></li>
<li><p><strong>Elana J. Fertig</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0003-3204-342X">0000-0003-3204-342X</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/ejfertig">ejfertig</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/FertigLab">FertigLab</a><br> <small> Department of Oncology, Sidney Kimmel Comprehensive Cancer Center, School of Medicine, Johns Hopkins University; Department of Applied Mathematics and Statistics, Whiting School of Engineering, Johns Hopkins University </small></p></li>
<li><p><strong>Michael I. Love</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0001-8401-0545">0000-0001-8401-0545</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/mikelove">mikelove</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/mikelove">mikelove</a><br> <small> Department of Biostatistics, University of North Carolina at Chapel Hill; Department of Genetics, University of North Carolina at Chapel Hill </small></p></li>
</ul>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<p><strong>Instructions</strong>: Describe your collaborative project, highlighting key achievements of the project; limited to 250 words.</p>
<h2 id="five-key-references">Five Key References</h2>
<ul>
<li>Hicks refs: <span class="citation" data-cites="DJaucmAA">[<a href="#ref-DJaucmAA">1</a>]</span></li>
<li>ProjectR &amp; scCoGAPS: <span class="citation" data-cites="cJPxOJMp">[<a href="#ref-cJPxOJMp">2</a>]</span></li>
<li>Alevin: <span class="citation" data-cites="FPpU83vH">[<a href="#ref-FPpU83vH">3</a>]</span></li>
</ul>
<h2 id="project-team">Project Team</h2>
<h3 id="pi-information">PI information</h3>
<ol type="1">
<li><p>Loyal Goff (Submitter)</p>
<ul>
<li>Title: Assistant Professor</li>
<li>Degrees: PhD</li>
<li>Type of organization: Academic</li>
<li>Tax ID: 52-0595110 (JHU)</li>
<li>Email: loyalgoff@jhmi.edu</li>
</ul></li>
<li><p>Stephanie Hicks</p>
<ul>
<li>Title: Assistant Professor</li>
<li>Degrees: PhD</li>
<li>Type of organization: Academic</li>
<li>Tax ID: 52-0595110 (JHU)</li>
<li>Email: shicks19@jhu.edu</li>
</ul></li>
<li><p>Elana Fertig</p>
<ul>
<li>Title: Associate Professor</li>
<li>Degrees: PhD</li>
<li>Type of organization: Academic</li>
<li>Tax ID: 52-0595110 (JHU)</li>
<li>Email: ejfertig@jhmi.edu</li>
</ul></li>
<li><p>Casey Greene</p>
<ul>
<li>Title: Assistant Professor</li>
<li>Degrees: PhD</li>
<li>Type of organization: Academic</li>
<li>Tax ID: 23-1352685 (UPenn)</li>
<li>Email: greenescientist@gmail.com</li>
</ul></li>
<li><p>Tom Hampton</p>
<ul>
<li>Title: Senior Bioinformatics Analyst</li>
<li>Degrees: PhD</li>
<li>Type of organization: Academic</li>
<li>Tax ID: 02-0222111 (Dartmouth)</li>
<li>Email: Thomas.H.Hampton@dartmouth.edu</li>
</ul></li>
<li><p>Michael Love</p>
<ul>
<li>Title: Assistant Professor</li>
<li>Degrees: Dr. rer. nat.</li>
<li>Type of organization: Academic</li>
<li>Tax ID: 56-6001393 (UNC)</li>
<li>Email: milove@email.unc.edu</li>
</ul></li>
<li>Rob Patro
<ul>
<li>Title: Assistant Professor</li>
<li>Degrees: PhD</li>
<li>Type of Organization: Academic</li>
<li>Tax ID: 16-1514621 (Stony Brook)</li>
<li>Email: rob.patro@cs.stonybrook.edu</li>
</ul></li>
</ol>
<h3 id="description-750-words-total">Description (750 words TOTAL)</h3>
<ol type="1">
<li>Loyal Goff</li>
<li>Stephanie C. Hicks is an Assistant Professor of Biostatistics at the Johns Hopkins Bloomberg School of Public Health. She is an expert in statistical methodology with a strong track record in processing and analyzing single-cell genomics data, including extensive experience developing fast, memory-efficient R/Bioconductor software to remove systematic and technical biases from scRNA-seq data <span class="citation" data-cites="DJaucmAA">[<a href="#ref-DJaucmAA">1</a>]</span>. Dr. Hicks will work together with Co-PIs to implement fast search algorithms in latent spaces (Aim 1) and to implement the methods developed into fast, scalable, and memory-efficient R/Bioconductor software packages (Aim 3).</li>
<li>Elana Fertig is an Associate Professor of Oncology and Applied Mathematics and Statistics at Johns Hopkins University. She developed of the Bayesian non-negative matrix factorization algorithm CoGAPS <span class="citation" data-cites="1DrhKLdVp">[<a href="#ref-1DrhKLdVp">4</a>]</span> for latent space analysis. In collaboration with co-PI Goff, she adapted this tool to scRNA-seq data and developed a new transfer learning framework to relate the low-dimensional features in scRNA-seq data across data modalities, biological conditions, and organisms <span class="citation" data-cites="cJPxOJMp">[<a href="#ref-cJPxOJMp">2</a>]</span>. Dr. Fertig will work with the co-PIs to incorporate the error models from Aim 1 into the latent space representations, dimensionality estimation, and biological assessment metrics in Aim 2. She is developing standardized language for latent space representation in collaboration with co-PIs Goff and Greene <span class="citation" data-cites="Sn52lYwa">[<a href="#ref-Sn52lYwa">5</a>]</span> that will provide a strong foundation for standardization of these approaches across different unsupervised learning tools.</li>
<li>Casey Greene</li>
<li>Tom Hampton</li>
<li>Michael Love is an Assistant Professor of Biostatistics and Genetics at the University of North Carolina at Chapel Hill. He is a leading developer of statistical software for RNA-seq analysis in the Bioconductor Project, maintaining the widely used DESeq2 <span class="citation" data-cites="w9AOzBMw">[<a href="#ref-w9AOzBMw">6</a>]</span> and tximport <span class="citation" data-cites="9CN5KEFo">[<a href="#ref-9CN5KEFo">7</a>]</span> packages. He is a close collaborator with Dr. Rob Patro on bias-aware estimation of transcript abundance from RNA-seq and estimation of uncertainty during transcript quantification <span class="citation" data-cites="vrqQcFyx">[<a href="#ref-vrqQcFyx">8</a>]</span>. Dr. Love will work with co-PIs to disseminate versioned reference cell type catalogs through widely used frameworks for genomic data analysis including R/Bioconductor and Python.</li>
<li>Rob Patro is an Assistant Professor of Computer Science at Stony Brook University. He leads the COMBINE-lab, that <a href="https://github.com/COMBINE-lab">develops and maintains numerous open-source genomics tools and methods</a>. He is the primary developer of the popular transcript quantification tools Sailfish <span class="citation" data-cites="RIPzCufe">[<a href="#ref-RIPzCufe">9</a>]</span> and Salmon <span class="citation" data-cites="vrqQcFyx">[<a href="#ref-vrqQcFyx">8</a>]</span>, having collaborated closely with Dr. Love on the latter. He and Dr. Love are actively collaborating on improved methods for transcript quantification, differential testing, and also on reproducible analysis via <a href="https://github.com/mikelove/tximeta">tximeta</a>. He has recently been focused on developing improved methods for gene-level quantification from tagged-end single-cell RNA-seq data, as implemented in the alevin tool <span class="citation" data-cites="FPpU83vH">[<a href="#ref-FPpU83vH">3</a>]</span>. Dr. Patro will worth with co-PIs to develop improved single-cell quantification tools that can account for gene-ambiguous reads and provide uncertainty information about the quantification estimates (base enabling technologies) — which is important for accurate and robust creation of reduced-dimensionality latent spaces. He will also work with the co-PIs to develop efficient algorithms and data structures, and robust implementations, to enable efficient expression search over low-dimensional representations of HCA data (Aim 1).</li>
</ol>
<h2 id="proposal-body-2000-words">Proposal Body (2000 words)</h2>
<p>The Human Cell Atlas (HCA) provides unprecedented characterization of molecular phenotypes across individuals, tissues and disease states – resolving differences to the level of individual cells. This dataset provides an extraordinary opportunity for scientific advancement enabled by new tools to rapidly query, characterize, and analyze these intrinsically high-dimensional data. To facilitate this, our seed network proposes to compress HCA data into fewer dimensions that preserve the important attributes of the original high dimensional data and yield interpretable, searchable features. For transcriptomic data, compressing on the gene dimension is most attractive: it can be applied to single samples, and genes often provide information about other co-regulated genes or cellular attributes. We hypothesize that building an ensemble of low dimensional representations across latent space methods will provide a<br />
reduced dimensional space that captures biological sources of variability that is robust to measurement noise. Our seed network will incorporate biologists and computer scientists from five leading academic institutions who will work together to create foundational technologies and educational opportunities that promote effective interpretation of low dimensional representations of HCA data. We will continue our active collaborations with other members of the broader HCA network to integrate state of the art latent space tools, portals, and annotations to enable biological utilization of the resulting low dimensional catalogue of HCA data.</p>
<h2 id="scientific-goals">Scientific Goals</h2>
<p>We will create low-dimensional representations that provide search and catalog capabilities for the HCA. Given both the scale of data, and the inherent complexity of biological systems, we believe these approaches are crucial to the long term success of the HCA. Our <strong><strong>central hypothesis</strong></strong> is that these approaches will enable faster algorithms while reducing the influence of technical noise. We propose to advance <strong><strong>base enabling technologies</strong></strong> for low-dimensional representations.</p>
<p>First, we will identify techniques that learn interpretable, biologically-aligned representations. We will consider both linear and non-linear techniques as each may identify distinct components of biological systems. For linear techniques, we rely on our Bayesian, non-negative matrix factorization method scCoGAPS <span class="citation" data-cites="6i1NIkNx">[<a href="#ref-6i1NIkNx">10</a>]</span> (PIs Fertig &amp; Goff). This technique learns biologically relevant features across contexts and data modalities <span class="citation" data-cites="wkhRfjyx">[<a href="#ref-wkhRfjyx">11</a>]</span>, including notably the HPN DREAM8 challenge <span class="citation" data-cites="qpg6x7P4">[<a href="#ref-qpg6x7P4">15</a>]</span>. This technique is specifically selected as a base enabling technnology because its error distribution can naturally account for measurement-specific technical variation <span class="citation" data-cites="5Cj8i4Xu">[<a href="#ref-5Cj8i4Xu">16</a>]</span> and its prior distributions for different feature quantifications or spatial information. For non-linear needs, neural networks with multiple layers, provide a complementary path to low-dimensional representations <span class="citation" data-cites="5CsWRjfp">[<a href="#ref-5CsWRjfp">17</a>]</span> (PI Greene) that model these diverse features of HCA data. We will make use of substantial progress that has already been made in both linear and non-linear techniques (e.g., <span class="citation" data-cites="vpa3pNZU">[<a href="#ref-vpa3pNZU">18</a>]</span>). and rigorously evaluate emerging methods into our search and catalog tools. We will extend transfer learning methods, including ProjectR <span class="citation" data-cites="cJPxOJMp">[<a href="#ref-cJPxOJMp">2</a>]</span> (PIs Goff &amp; Fertig) to enable rapid integration, interpretation, and annotation of learned latent spaces. The latent space team from the HCA collaborative networks RFA (including PIs Fertig, Goff, Greene, and Patro) is establishing common definitions and requirements for latent spaces for the HCA, as well as standardized output formats for low-dimensional representations from distinct classes of methods.</p>
<p>Second, we will improve techniques for fast and accurate quantification. Existing approaches for scRNA-seq data using tagged-end end protocols (e.g. 10x Chromium, drop-Seq, inDrop, etc.) do not account for reads mapping between multiple genes. This affects approximately 15-25% of the reads generated in a typical experiment, reducing quantification accuracy, and leads to systematic biases in gene expression estimates<span class="citation" data-cites="FPpU83vH">[<a href="#ref-FPpU83vH">3</a>]</span>. To address this, we will build on our recently developed quantification method for tagged-end data that accounts for reads mapping to multiple genomic loci in a principled and consistent way <span class="citation" data-cites="FPpU83vH">[<a href="#ref-FPpU83vH">3</a>]</span> (PI Patro) and extend this into a production quality tool for scRNA-Seq preprocessing. Our tool will support: 1. Exploration of alternative models for Unique Molecular Identifier (UMI) resolution. 2. Development of new approaches for quality control and filtering using the UMI-resolution graph. 3. Creation of a compressed and indexible data structure for the UMI-resolution graph to enable direct access, query, and fast search prior to secondary analysis.</p>
<p>We will implement these base enabling technologies and methods for search, analysis, and latent space transformations as freely available, open source software tools. We will additionally develop platform-agnostic input and output data formats and standards for latent space representations of the HCA data to maximize interoperability. The software tools produced will be fast, scalable, and memory-efficient by leveraging the available assets and expertises of the R/Bioconductor project (PIs Hicks &amp; Love) as well as the broader HCA community.</p>
<p>By using and extending our base enabling technologies we will provide three principle tools and resources for the HCA. These include 1) software to enable fast and accurate search and annotation using low-dimensional representations of cellular features, 2) a versioned and annotated catalog of latent spaces corresponding to signatures of cell types, states, and biological attributes across the the HCA, and 3) short course and educational materials that will increase the use and impact of low-dimensional representations and the HCA in general.</p>
<h3 id="aim-1">Aim 1</h3>
<p><em>Rationale:</em> The HCA provides a reference atlas to human cell types, states, and the biological processes in which they engage. The utility of the reference therefore requires that one can easily compare references to each other or a new sample to the compendium of reference samples. Low-dimensional representations, because they compress the space, provide the building blocks for search approaches that can be practically applied across very large datasets such as the HCA. <em>We propose to develop algorithms and software for efficient search over the HCA using low-dimensional representations.</em></p>
<p>The primary approach to search in low-dimensional spaces is straightforward: one must create an appropriate low-dimensional representation and identify distance functions that enable biologically meaningful comparisons. Ideal low-dimensional representations are predicted to be much faster to search, and potentially more biologically relevant, as noise can be removed. In this aim, we will evaluate novel low-dimensional representations to identify those with optimal qualities of compression, noise reduction, and retention of biologically meangful features. Current scRNA-Seq approaches require investigators to perform gene-level quantification on the entirety of a new sample. We aim to enable search during sample preprocessing, prior to gene-level quantification. This will enable in-line annotation of cell types and states and identification of novel features as samples are being processed. We will implement and evaluate techniques to learn and transfer shared low-dimensional representations between the UMI-resolution graph and quantified samples, so that samples where either component is available can be used for search and annotation <strong>[CASEY ADD SHARED LATENT SPACE REF]</strong>. These UMI-graphs will be embedded in the prior of scCoGAPS and architecture of non-linear latent space techniques. <strong>[Do we need this line? It’s a bit more specific than the rest of the paragraph -LAG]</strong> <strong>[I think we need something to link in how this fits to the latent space methods – maybe not so specific, but something that ties it back beyond preprocessing - EJF]</strong></p>
<p>Similarly to the approach by which comparisons to a reference genomes can identify specific differences in a genome of interest, we will use low-dimensional representations from latent spaces to define a reference transcriptome map (the HCA) and use this to quantify differences in target transcriptome maps from new samples of interest. We will leverage common low-dimensional representations and cell-to-cell correlation structure both within and across transcriptome maps from Aim 2 to define this reference. Quantifying the differences between samples characterized at the single-cell level reveals population or individual level differences. <strong>[&lt;– I’m not sure what this sentence means. Please clarify. - LAG]</strong> <strong>[My take is that it means if we have an average from the catalogue we’ve built for a cell type or state, that deviations in particular samples could yield context-specific differences, not sure how to reword - EJF]</strong> Comparison of scRNA-seq maps from individuals with a particular phenotype to the HCA reference that is computationally infeasible from the large scale of HCA data becomes tractable in these low dimensional spaces. We (PI Hicks) have extensive experience dealing with the distributions of cell expression within and between individuals <span class="citation" data-cites="13owodqhx">[<a href="#ref-13owodqhx">26</a>]</span>, which will be critical for defining an appropriate metric to compare references in latent spaces. We plan to implement and evaluate linear mixed models to account for the correlation structure within and between transcriptome maps. This statistical method will be fast, memory-efficient and will be scalable to billions of cells using low-dimensional representations.</p>
<h3 id="aim-2">Aim 2</h3>
<p><em>Rationale:</em> Biological systems are comprised of diverse cell types and states with overlapping molecular phenotypes. Furthermore, biological processes are often reused with modifications across cell types. Low-dimensional representations can identify these shared features, independent of total distance between cells in gene expression space, across large collections of data including the HCA. We will evaluate and select methods that define latent spaces that reflect discrete biological processes or cellular features. These latent spaces can be shared across different biological systems and can reveal context-specific divergence such as pathogenic differences in disease. <em>We propose to establish a central catalog of cell types, states, and biological processes derived from low-dimensional representations of the HCA.</em></p>
<p>By establishing a catalog of cellular features using low-dimensional representations can reduce noise and aid in biological interpretability. However, there are currently no standardized, quantitative metrics to determine the extent to which low-dimensional representations capture generalizable biolobical features. We have developed new transfer learning methods to quantify the extent to which latent space representations from one set of training data are represented in another <span class="citation" data-cites="cJPxOJMp">[<a href="#ref-cJPxOJMp">2</a>]</span> (PIs Goff &amp; Fertig). These provide a strong foundation to compare low-dimensional representations across different low-dimensional data representation technniques. Generalizable representations should transfer across datasets of related biological contexts, while representations of noise will not. In addition, we have found that combining multiple representations can better capture biological processes across scales <span class="citation" data-cites="Hlprh8TG">[<a href="#ref-Hlprh8TG">28</a>]</span>, and that representations across scales capture distinct, valid biological signatures <span class="citation" data-cites="5Cj8i4Xu">[<a href="#ref-5Cj8i4Xu">16</a>]</span>. Therefore, we will establish a versioned catalog consisting of low-dimensional features learned across both linear and non-linear methods from our base enabling technologies and proposed extensions in Aim 1.</p>
<p>We will package and version low-dimensional representations and annotate these representations based on their corresponding celluar features (e.g. cell type, tissue, biological process) and deliver these as structured data objects in Bioconductor as well as platform-agnostic data formats. Where applicable, we will leverage the computational tools previously developed by Bioconductor for single-cell data access to the HCA, data representation (<code>SingleCellExperiment</code>, <code>beachmat</code>, <code>LinearEmbeddingMatrix</code>, <code>DelayedArray</code>, <code>HDF5Array</code> and <code>rhdf5</code>) and data assessment and amelioration of data quality (<code>scater</code>, <code>scran</code>, <code>DropletUtils</code>). Such summaries and annotations have proven widely successful for the ENCODE, Roadmap Epigenome Mapping, and GTEx projects. We are core package developers and power users of Bioconductor (PIs Hicks and Love) and will support on-the-fly downloading of these materials via the <em>AnnotationHub</em> framework. To enable reproducible research leveraging HCA, we will implement a content-based versioning system, which identifies versions of the reference cell type catalog by the gene weights and transcript nucleotide sequences using a hash function. We (PI Love) developed hash-based versioning and provenance identification and detection framework for bulk RNA-seq that supports reproducible computational analyses and has proven to be successful <span class="citation" data-cites="1FQ0kp4Dj">[<a href="#ref-1FQ0kp4Dj">29</a>]</span>. This will help to avoid scenarios where researchers report on matches to a certain cell type in HCA without precisely defining which definition of that cell type. We will develop <em>F1000Research</em> workflows demonstrating how HCA-defined reference cell types and tools developed in this RFA can be used within a typical genomic data analysis. This catalogue will be used as the basis of defining the references for cell type and state, or individual-specific differences with the linear models proposed in Aim 1.</p>
<h3 id="aim-3">Aim 3</h3>
<p><em>Rationale:</em> Low-dimensional representations of scRNA-seq and HCA data make tasks faster and provide interpretable summaries of complex high-dimensional cellular features. The HCA data-associated methods and workflows will be valuable to many biomedical fields, but their use will require an understanding of basic bioinformatics, scRNA-Seq, and how the tools being developed work. Furthermore, researchers will need exposure to the conceptual basis of low-dimensional interpretations of biological systems. This aim addresses these needs in three ways.</p>
<p>First, we will develop a bioinformatic training program for biologists at all levels, including those with no experience in bioinformatics. Lecture materials will be extended from existing materials from previous bioinformatic courses we (PI Hampton) have run at Mount Desert Island Biological Laboratory, the University of Birmingham, UK, and Geisel School of Medicine at Dartmouth since 2009. These courses have trained over 400 scientists in basic bioinformatics and always achieve approval ratings of over 90%. We believe part of the success of these learning experiences has to do with our instructional paradigm, which includes a very challenging course project coupled with one-on-one support from instructors. We will develop a new curriculum specifically tailored to HCA that incorporates: 1) didactic course material on single cell gene expression profiling previously developed as part of the annual McKusick Short Course on Human and Mammalian Genetics at Jackson Labs (PI Goff), 2) machine learning methods (PI Greene), 3) search and analysis in low-dimensional representations, and 4) tools developed by our group in response to this RFA.</p>
<p>Second, the short course will train not only students, but instructors. Our one-on-one approach to course projects will require a high instructor to student ratio. We will therefore recruit former participants of this class to return in subsequent years, first as teaching assistants, and later as module presenters. We have found that course alumni are eager to improve their teaching resumes, that they learn the material in a new way as they begin to teach it, and that they are an invaluable resource in understanding how to improve the course over time. Part of our strategy is to support this community, which includes many people who will drive the next wave of innovation. All of our course materials will be freely available, enabling course participants to bring what they learned home with them. A capstone session will be included in which we will provide suggestions about how the materials presented in the course can be incorporated into existing course curricula. Course faculty will be available to assist with integration effort after the course.</p>
<p>Finally, the short course will facilitate scientific collaborations as follows. In addition to engaging in group presentations as part of the class project that will involve scRNA-seq data retrieved from HCA, participants will also provide a 10-minute chalk talk explaining their research interests. This allows course faculty to identify course participants whose needs they can help address, both during and after the course. The course design will include faculty office hours where participants can get help with any bioinformatic problems they are encountering with their ongoing research.</p>
<p>We will provide this course free to 25 students twice a year, with the understanding that its effective reach will be much greater than the 150 students who enroll. Our students will begin to teach others, and our materials will be incorporated into many local courses.</p>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-DJaucmAA">
<p>1. <strong>Missing data and technical variability in single-cell RNA-sequencing experiments</strong><br />
Stephanie C Hicks, F William Townes, Mingxiang Teng, Rafael A Irizarry<br />
<br />
<em>Biostatistics</em> (2017-11-06) <a href="https://doi.org/gfb8g4" class="uri">https://doi.org/gfb8g4</a><br />
DOI: <a href="https://doi.org/10.1093/biostatistics/kxx053">10.1093/biostatistics/kxx053</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/29121214">29121214</a></p>
</div>
<div id="ref-cJPxOJMp">
<p>2. <strong>Decomposing cell identity for transfer learning across cellular measurements, platforms, tissues, and species.</strong><br />
Genevieve L Stein-O’Brien, Brian S. Clark, Thomas Sherman, Christina Zibetti, Qiwen Hu, Rachel Sealfon, Sheng Liu, Jiang Qian, Carlo Colantuoni, Seth Blackshaw, … Elana J. Fertig<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-08-20) <a href="https://doi.org/gd2xpn" class="uri">https://doi.org/gd2xpn</a><br />
DOI: <a href="https://doi.org/10.1101/395004">10.1101/395004</a></p>
</div>
<div id="ref-FPpU83vH">
<p>3. <strong>Alevin efficiently estimates accurate gene abundances from dscRNA-seq data</strong><br />
Avi Srivastava, Laraib Malik, Tom Sean Smith, Ian Sudbery, Rob Patro<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-06-01) <a href="https://doi.org/gffk42" class="uri">https://doi.org/gffk42</a><br />
DOI: <a href="https://doi.org/10.1101/335000">10.1101/335000</a></p>
</div>
<div id="ref-1DrhKLdVp">
<p>4. <strong>CoGAPS: an R/C++ package to identify patterns and biological process activity in transcriptomic data</strong><br />
Elana J. Fertig, Jie Ding, Alexander V. Favorov, Giovanni Parmigiani, Michael F. Ochs<br />
<br />
<em>Bioinformatics</em> (2010-09-01) <a href="https://doi.org/cwqsv4" class="uri">https://doi.org/cwqsv4</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btq503">10.1093/bioinformatics/btq503</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20810601">20810601</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3025742">PMC3025742</a></p>
</div>
<div id="ref-Sn52lYwa">
<p>5. <strong>Enter the Matrix: Factorization Uncovers Knowledge from Omics</strong><br />
Genevieve L. Stein-O’Brien, Raman Arora, Aedin C. Culhane, Alexander V. Favorov, Lana X. Garmire, Casey S. Greene, Loyal A. Goff, Yifeng Li, Aloune Ngom, Michael F. Ochs, … Elana J. Fertig<br />
<br />
<em>Trends in Genetics</em> (2018-10) <a href="https://doi.org/gd93tk" class="uri">https://doi.org/gd93tk</a><br />
DOI: <a href="https://doi.org/10.1016/j.tig.2018.07.003">10.1016/j.tig.2018.07.003</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/30143323">30143323</a></p>
</div>
<div id="ref-w9AOzBMw">
<p>6. <strong>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</strong><br />
Michael I Love, Wolfgang Huber, Simon Anders<br />
<br />
<em>Genome Biology</em> (2014-12) <a href="https://doi.org/gd3zvn" class="uri">https://doi.org/gd3zvn</a><br />
DOI: <a href="https://doi.org/10.1186/s13059-014-0550-8">10.1186/s13059-014-0550-8</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25516281">25516281</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4302049">PMC4302049</a></p>
</div>
<div id="ref-9CN5KEFo">
<p>7. <strong>Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences</strong><br />
Charlotte Soneson, Michael I. Love, Mark D. Robinson<br />
<br />
<em>F1000Research</em> (2015-12-30) <a href="https://doi.org/gdtgw8" class="uri">https://doi.org/gdtgw8</a><br />
DOI: <a href="https://doi.org/10.12688/f1000research.7563.1">10.12688/f1000research.7563.1</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26925227">26925227</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4712774">PMC4712774</a></p>
</div>
<div id="ref-vrqQcFyx">
<p>8. <strong>Salmon provides fast and bias-aware quantification of transcript expression</strong><br />
Rob Patro, Geet Duggal, Michael I Love, Rafael A Irizarry, Carl Kingsford<br />
<br />
<em>Nature Methods</em> (2017-03-06) <a href="https://doi.org/gcw9f5" class="uri">https://doi.org/gcw9f5</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4197">10.1038/nmeth.4197</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28263959">28263959</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5600148">PMC5600148</a></p>
</div>
<div id="ref-RIPzCufe">
<p>9. <strong>Sailfish enables alignment-free isoform quantification from RNA-seq reads using lightweight algorithms</strong><br />
Rob Patro, Stephen M Mount, Carl Kingsford<br />
<br />
<em>Nature Biotechnology</em> (2014-04-20) <a href="https://doi.org/gfghc2" class="uri">https://doi.org/gfghc2</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.2862">10.1038/nbt.2862</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24752080">24752080</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4077321">PMC4077321</a></p>
</div>
<div id="ref-6i1NIkNx">
<p>10. <strong>Comprehensive analysis of retinal development at single cell resolution identifies NFI factors as essential for mitotic exit and specification of late-born cells</strong><br />
Brian Clark, Genevieve Stein-O’Brien, Fion Shiau, Gabrielle Cannon, Emily Davis, Thomas Sherman, Fatemeh Rajaii, Rebecca James-Esposito, Richard Gronostajski, Elana Fertig, … Seth Blackshaw<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-07-27) <a href="https://doi.org/gdwrzh" class="uri">https://doi.org/gdwrzh</a><br />
DOI: <a href="https://doi.org/10.1101/378950">10.1101/378950</a></p>
</div>
<div id="ref-wkhRfjyx">
<p>11. <strong>Gene expression signatures modulated by epidermal growth factor receptor activation and their relationship to cetuximab resistance in head and neck squamous cell carcinoma</strong><br />
Elana J Fertig, Qing Ren, Haixia Cheng, Hiromitsu Hatakeyama, Adam P Dicker, Ulrich Rodeck, Michael Considine, Michael F Ochs, Christine H Chung<br />
<br />
<em>BMC Genomics</em> (2012) <a href="https://doi.org/gb3fgp" class="uri">https://doi.org/gb3fgp</a><br />
DOI: <a href="https://doi.org/10.1186/1471-2164-13-160">10.1186/1471-2164-13-160</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22549044">22549044</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3460736">PMC3460736</a></p>
</div>
<div id="ref-1GkdWBzqU">
<p>12. <strong>CoGAPS matrix factorization algorithm identifies transcriptional changes in AP-2alpha target genes in feedback from therapeutic inhibition of the EGFR network</strong><br />
Elana J. Fertig, Hiroyuki Ozawa, Manjusha Thakar, Jason D. Howard, Luciane T. Kagohara, Gabriel Krigsfeld, Ruchira S. Ranaweera, Robert M. Hughes, Jimena Perez, Siân Jones, … Christine H. Chung<br />
<br />
<em>Oncotarget</em> (2016-09-16) <a href="https://doi.org/f9k8d8" class="uri">https://doi.org/f9k8d8</a><br />
DOI: <a href="https://doi.org/10.18632/oncotarget.12075">10.18632/oncotarget.12075</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/27650546">27650546</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5342018">PMC5342018</a></p>
</div>
<div id="ref-uInnOMwX">
<p>13. <strong>Pattern Identification in Time-Course Gene Expression Data with the CoGAPS Matrix Factorization</strong><br />
Elana J. Fertig, Genevieve Stein-O’Brien, Andrew Jaffe, Carlo Colantuoni<br />
<br />
<em>Gene Function Analysis</em> (2013-10-24) <a href="https://doi.org/f5j7xj" class="uri">https://doi.org/f5j7xj</a><br />
DOI: <a href="https://doi.org/10.1007/978-1-62703-721-1_6">10.1007/978-1-62703-721-1_6</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24233779">24233779</a></p>
</div>
<div id="ref-1DZRsfkoC">
<p>14. <strong>Integrated time course omics analysis distinguishes immediate therapeutic response from acquired resistance</strong><br />
Genevieve Stein-O’Brien, Luciane T. Kagohara, Sijia Li, Manjusha Thakar, Ruchira Ranaweera, Hiroyuki Ozawa, Haixia Cheng, Michael Considine, Sandra Schmitz, Alexander V. Favorov, … Elana J. Fertig<br />
<br />
<em>Genome Medicine</em> (2018-05-23) <a href="https://doi.org/gfc4dq" class="uri">https://doi.org/gfc4dq</a><br />
DOI: <a href="https://doi.org/10.1186/s13073-018-0545-2">10.1186/s13073-018-0545-2</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/29792227">29792227</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5966898">PMC5966898</a></p>
</div>
<div id="ref-qpg6x7P4">
<p>15. <strong>Inferring causal molecular networks: empirical assessment through a community-based effort</strong><br />
Steven M HillLaura M Heiser, Thomas Cokelaer, Michael Unger, Nicole K Nesser, Daniel E Carlin, Yang Zhang, Artem Sokolov, Evan O Paull, … Sach Mukherjee<br />
<br />
<em>Nature Methods</em> (2016-02-22) <a href="https://doi.org/f3t7t4" class="uri">https://doi.org/f3t7t4</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.3773">10.1038/nmeth.3773</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26901648">26901648</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4854847">PMC4854847</a></p>
</div>
<div id="ref-5Cj8i4Xu">
<p>16. <strong>Preferential Activation of the Hedgehog Pathway by Epigenetic Modulations in HPV Negative HNSCC Identified with Meta-Pathway Analysis</strong><br />
Elana J. Fertig, Ana Markovic, Ludmila V. Danilova, Daria A. Gaykalova, Leslie Cope, Christine H. Chung, Michael F. Ochs, Joseph A. Califano<br />
<br />
<em>PLoS ONE</em> (2013-11-04) <a href="https://doi.org/gcpgc6" class="uri">https://doi.org/gcpgc6</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pone.0078127">10.1371/journal.pone.0078127</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24223768">24223768</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3817178">PMC3817178</a></p>
</div>
<div id="ref-5CsWRjfp">
<p>17. <strong>Parameter tuning is a key part of dimensionality reduction via deep variational autoencoders for single cell RNA transcriptomics</strong><br />
Qiwen Hu, Casey S Greene<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-08-05) <a href="https://doi.org/gdxxjf" class="uri">https://doi.org/gdxxjf</a><br />
DOI: <a href="https://doi.org/10.1101/385534">10.1101/385534</a></p>
</div>
<div id="ref-vpa3pNZU">
<p>18. <strong>Single cell RNA-seq denoising using a deep count autoencoder</strong><br />
Gökcen Eraslan, Lukas M. Simon, Maria Mircea, Nikola S. Mueller, Fabian J. Theis<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-04-13) <a href="https://doi.org/gdjcb3" class="uri">https://doi.org/gdjcb3</a><br />
DOI: <a href="https://doi.org/10.1101/300681">10.1101/300681</a></p>
</div>
<div id="ref-XjfRUvN5">
<p>19. <strong>Bayesian Inference for a Generative Model of Transcriptome Profiles from Single-cell RNA Sequencing</strong><br />
Romain Lopez, Jeffrey Regier, Michael B Cole, Michael I Jordan, Nir Yosef<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-03-30) <a href="https://doi.org/gdm9jf" class="uri">https://doi.org/gdm9jf</a><br />
DOI: <a href="https://doi.org/10.1101/292037">10.1101/292037</a></p>
</div>
<div id="ref-TkR5VPF7">
<p>20. <strong>Exploring Single-Cell Data with Deep Multitasking Neural Networks</strong><br />
Matthew Amodio, David van Dijk, Krishnan Srinivasan, William S Chen, Hussein Mohsen, Kevin R Moon, Allison Campbell, Yujiao Zhao, Xiaomei Wang, Manjunatha Venkataswamy, … Smita Krishnaswamy<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2017-12-19) <a href="https://doi.org/gfgrpk" class="uri">https://doi.org/gfgrpk</a><br />
DOI: <a href="https://doi.org/10.1101/237065">10.1101/237065</a></p>
</div>
<div id="ref-V3nGUaio">
<p>21. <strong>Massive single-cell RNA-seq analysis and imputation via deep learning</strong><br />
Yue Deng, Feng Bao, Qionghai Dai, Lani Wu, Steven Altschuler<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-05-06) <a href="https://doi.org/gfgrpm" class="uri">https://doi.org/gfgrpm</a><br />
DOI: <a href="https://doi.org/10.1101/315556">10.1101/315556</a></p>
</div>
<div id="ref-eMe9qeSH">
<p>22. <strong>Transfer learning in single-cell transcriptomics improves data denoising and pattern discovery</strong><br />
Jingshu Wang, Divyansh Agarwal, Mo Huang, Gang Hu, Zilu Zhou, Vincent B. Conley, Hugh MacMullan, Nancy R. Zhang<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-10-31) <a href="https://doi.org/gfgrpn" class="uri">https://doi.org/gfgrpn</a><br />
DOI: <a href="https://doi.org/10.1101/457879">10.1101/457879</a></p>
</div>
<div id="ref-1GR3FIJMG">
<p>23. <strong>Efficient Generation of Transcriptomic Profiles by Random Composite Measurements</strong><br />
Brian Cleary, Le Cong, Anthea Cheung, Eric S. Lander, Aviv Regev<br />
<br />
<em>Cell</em> (2017-11) <a href="https://doi.org/gcrjhc" class="uri">https://doi.org/gcrjhc</a><br />
DOI: <a href="https://doi.org/10.1016/j.cell.2017.10.023">10.1016/j.cell.2017.10.023</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/29153835">29153835</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5726792">PMC5726792</a></p>
</div>
<div id="ref-GC2u23Xj">
<p>24. <strong>Detecting heterogeneity in single-cell RNA-Seq data by non-negative matrix factorization</strong><br />
Xun Zhu, Travers Ching, Xinghua Pan, Sherman M. Weissman, Lana Garmire<br />
<br />
<em>PeerJ</em> (2017-01-19) <a href="https://doi.org/gfgr7c" class="uri">https://doi.org/gfgr7c</a><br />
DOI: <a href="https://doi.org/10.7717/peerj.2888">10.7717/peerj.2888</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28133571">28133571</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5251935">PMC5251935</a></p>
</div>
<div id="ref-OekvE5up">
<p>25. <strong>Integrative inference of brain cell similarities and differences from single-cell genomics</strong><br />
Joshua Welch, Velina Kozareva, Ashley Ferreira, Charles Vanderburg, Carly Martin, Evan Macosko<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-11-02) <a href="https://doi.org/gfgr7b" class="uri">https://doi.org/gfgr7b</a><br />
DOI: <a href="https://doi.org/10.1101/459891">10.1101/459891</a></p>
</div>
<div id="ref-13owodqhx">
<p>26. <strong>quantro: a data-driven approach to guide the choice of an appropriate normalization method.</strong><br />
Stephanie C Hicks, Rafael A Irizarry<br />
<br />
<em>Genome biology</em> (2015-06-04) <a href="https://www.ncbi.nlm.nih.gov/pubmed/26040460" class="uri">https://www.ncbi.nlm.nih.gov/pubmed/26040460</a><br />
DOI: <a href="https://doi.org/10.1186/s13059-015-0679-0">10.1186/s13059-015-0679-0</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26040460">26040460</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495646">PMC4495646</a></p>
</div>
<div id="ref-1GtRgPRxn">
<p>27. <strong>MultiPLIER: a transfer learning framework reveals systemic features of rare autoimmune disease</strong><br />
Jaclyn N Taroni, Peter C Grayson, Qiwen Hu, Sean Eddy, Matthias Kretzler, Peter A Merkel, Casey S Greene<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-08-20) <a href="https://doi.org/gfc9bb" class="uri">https://doi.org/gfc9bb</a><br />
DOI: <a href="https://doi.org/10.1101/395947">10.1101/395947</a></p>
</div>
<div id="ref-Hlprh8TG">
<p>28. <strong>Unsupervised Extraction of Stable Expression Signatures from Public Compendia with an Ensemble of Neural Networks</strong><br />
Jie Tan, Georgia Doing, Kimberley A. Lewis, Courtney E. Price, Kathleen M. Chen, Kyle C. Cady, Barret Perchuk, Michael T. Laub, Deborah A. Hogan, Casey S. Greene<br />
<br />
<em>Cell Systems</em> (2017-07) <a href="https://doi.org/gcw9f4" class="uri">https://doi.org/gcw9f4</a><br />
DOI: <a href="https://doi.org/10.1016/j.cels.2017.06.003">10.1016/j.cels.2017.06.003</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28711280">28711280</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5532071">PMC5532071</a></p>
</div>
<div id="ref-1FQ0kp4Dj">
<p>29. <strong>tximeta</strong><br />
Rob Patro Michael Love<br />
<br />
<em>Bioconductor</em> (2018) <a href="https://doi.org/gfddxw" class="uri">https://doi.org/gfddxw</a><br />
DOI: <a href="https://doi.org/10.18129/b9.bioc.tximeta">10.18129/b9.bioc.tximeta</a></p>
</div>
</div>
<script>
// AnchorJS minified version below.
// Source https://github.com/bryanbraun/anchorjs/blob/064abdd0987f305933ec4982af6d0c1cf2fd0814/anchor.js

/**
 * AnchorJS - v4.0.0 - 2017-06-02
 * https://github.com/bryanbraun/anchorjs
 * Copyright (c) 2017 Bryan Braun; Licensed MIT
 */
!function(A,e){"use strict";"function"==typeof define&&define.amd?define([],e):"object"==typeof module&&module.exports?module.exports=e():(A.AnchorJS=e(),A.anchors=new A.AnchorJS)}(this,function(){"use strict";function A(A){function e(A){A.icon=A.hasOwnProperty("icon")?A.icon:"",A.visible=A.hasOwnProperty("visible")?A.visible:"hover",A.placement=A.hasOwnProperty("placement")?A.placement:"right",A.class=A.hasOwnProperty("class")?A.class:"",A.truncate=A.hasOwnProperty("truncate")?Math.floor(A.truncate):64}function t(A){var e;if("string"==typeof A||A instanceof String)e=[].slice.call(document.querySelectorAll(A));else{if(!(Array.isArray(A)||A instanceof NodeList))throw new Error("The selector provided to AnchorJS was invalid.");e=[].slice.call(A)}return e}function n(){if(null===document.head.querySelector("style.anchorjs")){var A,e=document.createElement("style");e.className="anchorjs",e.appendChild(document.createTextNode("")),void 0===(A=document.head.querySelector('[rel="stylesheet"], style'))?document.head.appendChild(e):document.head.insertBefore(e,A),e.sheet.insertRule(" .anchorjs-link {   opacity: 0;   text-decoration: none;   -webkit-font-smoothing: antialiased;   -moz-osx-font-smoothing: grayscale; }",e.sheet.cssRules.length),e.sheet.insertRule(" *:hover > .anchorjs-link, .anchorjs-link:focus  {   opacity: 1; }",e.sheet.cssRules.length),e.sheet.insertRule(" [data-anchorjs-icon]::after {   content: attr(data-anchorjs-icon); }",e.sheet.cssRules.length),e.sheet.insertRule(' @font-face {   font-family: "anchorjs-icons";   src: url(data:n/a;base64,AAEAAAALAIAAAwAwT1MvMg8yG2cAAAE4AAAAYGNtYXDp3gC3AAABpAAAAExnYXNwAAAAEAAAA9wAAAAIZ2x5ZlQCcfwAAAH4AAABCGhlYWQHFvHyAAAAvAAAADZoaGVhBnACFwAAAPQAAAAkaG10eASAADEAAAGYAAAADGxvY2EACACEAAAB8AAAAAhtYXhwAAYAVwAAARgAAAAgbmFtZQGOH9cAAAMAAAAAunBvc3QAAwAAAAADvAAAACAAAQAAAAEAAHzE2p9fDzz1AAkEAAAAAADRecUWAAAAANQA6R8AAAAAAoACwAAAAAgAAgAAAAAAAAABAAADwP/AAAACgAAA/9MCrQABAAAAAAAAAAAAAAAAAAAAAwABAAAAAwBVAAIAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAMCQAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAg//0DwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAAIAAAACgAAxAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADAAAAAIAAgAAgAAACDpy//9//8AAAAg6cv//f///+EWNwADAAEAAAAAAAAAAAAAAAAACACEAAEAAAAAAAAAAAAAAAAxAAACAAQARAKAAsAAKwBUAAABIiYnJjQ3NzY2MzIWFxYUBwcGIicmNDc3NjQnJiYjIgYHBwYUFxYUBwYGIwciJicmNDc3NjIXFhQHBwYUFxYWMzI2Nzc2NCcmNDc2MhcWFAcHBgYjARQGDAUtLXoWOR8fORYtLTgKGwoKCjgaGg0gEhIgDXoaGgkJBQwHdR85Fi0tOAobCgoKOBoaDSASEiANehoaCQkKGwotLXoWOR8BMwUFLYEuehYXFxYugC44CQkKGwo4GkoaDQ0NDXoaShoKGwoFBe8XFi6ALjgJCQobCjgaShoNDQ0NehpKGgobCgoKLYEuehYXAAAADACWAAEAAAAAAAEACAAAAAEAAAAAAAIAAwAIAAEAAAAAAAMACAAAAAEAAAAAAAQACAAAAAEAAAAAAAUAAQALAAEAAAAAAAYACAAAAAMAAQQJAAEAEAAMAAMAAQQJAAIABgAcAAMAAQQJAAMAEAAMAAMAAQQJAAQAEAAMAAMAAQQJAAUAAgAiAAMAAQQJAAYAEAAMYW5jaG9yanM0MDBAAGEAbgBjAGgAbwByAGoAcwA0ADAAMABAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAH//wAP) format("truetype"); }',e.sheet.cssRules.length)}}this.options=A||{},this.elements=[],e(this.options),this.isTouchDevice=function(){return!!("ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch)},this.add=function(A){var i,o,s,c,r,a,h,l,u,d,f,g,p=[];if(e(this.options),"touch"===(g=this.options.visible)&&(g=this.isTouchDevice()?"always":"hover"),A||(A="h2, h3, h4, h5, h6"),0===(i=t(A)).length)return this;for(n(),o=document.querySelectorAll("[id]"),s=[].map.call(o,function(A){return A.id}),r=0;r<i.length;r++)if(this.hasAnchorJSLink(i[r]))p.push(r);else{if(i[r].hasAttribute("id"))c=i[r].getAttribute("id");else if(i[r].hasAttribute("data-anchor-id"))c=i[r].getAttribute("data-anchor-id");else{u=l=this.urlify(i[r].textContent),h=0;do{void 0!==a&&(u=l+"-"+h),a=s.indexOf(u),h+=1}while(-1!==a);a=void 0,s.push(u),i[r].setAttribute("id",u),c=u}d=c.replace(/-/g," "),(f=document.createElement("a")).className="anchorjs-link "+this.options.class,f.href="#"+c,f.setAttribute("aria-label","Anchor link for: "+d),f.setAttribute("data-anchorjs-icon",this.options.icon),"always"===g&&(f.style.opacity="1"),""===this.options.icon&&(f.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(f.style.lineHeight="inherit")),"left"===this.options.placement?(f.style.position="absolute",f.style.marginLeft="-1em",f.style.paddingRight="0.5em",i[r].insertBefore(f,i[r].firstChild)):(f.style.paddingLeft="0.375em",i[r].appendChild(f))}for(r=0;r<p.length;r++)i.splice(p[r]-r,1);return this.elements=this.elements.concat(i),this},this.remove=function(A){for(var e,n,i=t(A),o=0;o<i.length;o++)(n=i[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(i[o]))&&this.elements.splice(e,1),i[o].removeChild(n));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}return A});

// Enable links for selected headers
var anchors = new AnchorJS();
anchors.add("h2, h3, h4")
</script>
<!-- Enable Hypothesis annotation. https://web.hypothes.is/ -->
<script src="https://hypothes.is/embed.js" async></script>
</body>
</html>
